### PR TITLE
BET-1

### PR DESCRIPTION
**Pull request by ahinchliff**
Looking for feedback on my solidity and testing :D

### CODE REVIEW

File: contracts/Pavilion.sol [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        string name;
+        Direction outcome;
+        uint256 closesAt;
+        Status status;
+        DirectionStruct prices;
+        DirectionStruct ticketsOutstanding;
+        DirectionStruct maxQuantity;
+        mapping(address => DirectionStruct) balances;
+        mapping(address => uint256) totalSpentInCents;
+    }
```
Comment (bitcoinbrisbane) [inline contracts/Pavilion.sol]: I solidity 8 you can put the structs outside of the contract.

File: contracts/Pavilion.sol [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+    function createGame(
+        string calldata name,
+        uint256 closesAt,
+        uint256 positivePrice,
+        uint256 negativePrice,
+        uint256 positiveMaxQuantity,
+        uint256 negativeMaxQuantity
+    ) public onlyOwner {
+        lastGameId += 1;
```
Comment (bitcoinbrisbane) [inline contracts/Pavilion.sol]:
```suggestion
        lastGameId ++;
```

File: contracts/Pavilion.sol [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        Game storage game = games[lastGameId];
+
+        game.id = lastGameId;
+        game.closesAt = closesAt;
```
Comment (bitcoinbrisbane) [inline contracts/Pavilion.sol:63]: Might want to do a require such as

```
require(closeAt > block.time, "Should be in the future");
```

**Review by rellfy — APPROVED** [review]
Looks good!

**Pull request merged**